### PR TITLE
Super Cache - tests round 2

### PR DESCRIPTION
**Pull request by thingalon**
This PR adds a few extra tests for Super Cache, and improves the reliability of the suite

#### Changes proposed in this Pull Request:
* Add new tests for compressing cache files
* Improve the reliability of tests by ensuring cleaner / better resets between tests.

#### Other information:
- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?

#### Jetpack product discussion
n/a

#### Does this pull request change what data or activity we track or use?
no

#### Testing instructions:
* `cd projects/plugins/super-cache/tests/e2e`
* `pnpm i`
* `pnpm env:up` (then wait a minute for the docker to finish spinning up)
* `pnpm test:run`

### CODE REVIEW

**Comment by haqadn** [comment]
Tests are failing on my machine with error:

```
Command failed: docker exec -u www-data super-cache-e2e_wordpress_1 wp plugin deactivate wp-super-cache --skip-themes
    Error: No such container: super-cache-e2e_wordpress_1
```

The docker container name for me is: `super-cache-e2e-wordpress-1`

I wonder if something changed for docker desktop that generates different container names. My docker desktop version is 4.12.0 (85629)

File: projects/plugins/super-cache/tests/e2e/specs/default-settings.test.ts [hunk 1]
```diff
@@ -13,21 +20,25 @@ describe( 'cache behavior with default settings', () => {
...
-	test( 'logged in users do not get cached pages', async () => {
+	test( 'logged in users get cached pages', async () => {
 		const cookie = await getAuthCookie();
 		const url = getSiteUrl();
 
 		const first = await authenticatedRequest( cookie, 'GET', url );
 		const second = await authenticatedRequest( cookie, 'GET', url );
 
-		expect( first ).not.toBe( second );
+		expect( first ).toBe( second );
```
Comment (haqadn) [inline projects/plugins/super-cache/tests/e2e/specs/default-settings.test.ts:41]: I am a bit confused about this test. Isn't it the default behavior not to cache logged-in users?

Comment (thingalon) [reply projects/plugins/super-cache/tests/e2e/specs/default-settings.test.ts:41]: It depends on how you turn Super Cache on. If you turn it on via the "Easy" tab, then yes it sets that setting by default. If you turn it on via the Advanced tab, it doesn't change any settings - including leaving caching set to "Enable caching for all visitors."

**Comment by thingalon** [comment]
> The docker container name for me is: super-cache-e2e-wordpress-1
> I wonder if something changed for docker desktop that generates different container names. My docker desktop version is 4.12.0 (85629)

Thanks for pointing this out. In https://github.com/Automattic/jetpack/pull/26662/commits/6b3748989b724a90af156e10740de89a49bc8abb I added code to detect the id of the container, so it should fix the issue.

**Review by pyronaur — APPROVED** [review]

**Pull request merged**